tutorials/scripting/gdextension/gdextension_docs_generation.rst

```diff
@@ -0,0 +1,47 @@
+.. _doc_gdextension_docs_generation:
+
+Adding documentation to your GDExtension
+========================================
+
+.. note::
+    Adding documentation for GDExtensions is only possible for 4.3 and later.
+
+Similarly to how the engine works with built-in documentation inside the editor is how the
+generation of the GDExtension documentation works.
+
+Inside the Godot project directory of your GDExtension directory you run the following terminal command:
+
+.. code-block:: none
+
+    godot --doctool ../ --gdextension-docs
+
+This command calls upon the Godot editor binary to generate documentation via the ``--doctool``
+and ``--gdextension-docs`` commands. The ``../`` addition is to let Godot know where the GDExtension
+SConstruct file is located. By calling this command Godot generates a ``doc_classes`` directory inside the
+project directory in which it generates ``xml`` files for the GDExtension classes. Those files
```

Comment:
My gut feeling demands this should be either "XML files" or "`.xml` files", as "xml files" feels like a weird in-between. However, not super important to me, and I haven't checked other places if there is any precedent for this.

Reply:
I think we had ".xml" files but I will check and rework this 👍🏻